AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/CMCs/StormTracker.dfy

```diff
@@ -139,34 +143,46 @@ module {:options "/functionSyntax:4" }  StormTracker {
         && fresh(this.wrapped)
         && fresh(this.wrapped.InternalModifies)
         && fresh(this.inFlight)
+      ensures
+        && inFlight.Size() == 0
+        && lastPrune == 0
     {
+      var gracePeriod, graceInterval, inFlightTTL;
+      if cache.timeUnits.UnwrapOr(Types.Seconds).Seconds? {
+        gracePeriod := cache.gracePeriod as Types.PositiveLong * 1000;
+        graceInterval := cache.graceInterval as Types.PositiveLong * 1000;
+        inFlightTTL := cache.inFlightTTL as Types.PositiveLong * 1000;
+      } else {
+        gracePeriod := cache.gracePeriod as Types.PositiveLong;
+        graceInterval := cache.graceInterval as Types.PositiveLong;
+        inFlightTTL := cache.inFlightTTL as Types.PositiveLong;
+      }
+
       this.wrapped := new LocalCMC.LocalCMC(cache.entryCapacity as nat, cache.entryPruningTailSize.UnwrapOr(1) as nat);
       this.inFlight := new MutableMap();
-      this.gracePeriod := cache.gracePeriod as Types.PositiveLong;
-      this.graceInterval := cache.graceInterval as Types.PositiveLong;
+      this.gracePeriod := gracePeriod;
+      this.graceInterval := graceInterval;
       this.fanOut := cache.fanOut as Types.PositiveLong;
-      this.inFlightTTL := cache.inFlightTTL as Types.PositiveLong;
+      this.inFlightTTL := inFlightTTL;
       this.sleepMilli := cache.sleepMilli as Types.PositiveLong;
       this.lastPrune := 0;
     }
 
-    function InFlightSize() : Types.PositiveLong
-      reads this, this.inFlight
-    {
-      var x := inFlight.Size();
-      assert x >= 0;
-      if x <= INT64_MAX_LIMIT then
-        x as Types.PositiveLong
-      else
-        INT64_MAX_LIMIT as Types.PositiveLong
-    }
-
     // return true if InFlight is full
     method FanOutReached(now : Types.PositiveLong) returns (res : bool)
       modifies this`lastPrune, inFlight
+      requires ValidState()
+      ensures ValidState()
+      ensures
+        && old(inFlight.Size()) < fanOut as nat
+        ==>
+          && !res
+          && inFlight.content() == old(inFlight.content())
+      ensures !res ==> inFlight.Size() < fanOut as nat
+      ensures res ==> inFlight.Size() == fanOut as nat
```

Comment:
Not important, but `==>` could be `<==>`

Reply:
Yup. Not too helpful as I have an `old` problem. Updated.